File: tennis-game/objective-c-tennis-game/objective-c-tennis-game/GameFactory.m
//
//  GameFactory.m
//  objective-c-tennis-game
//
//  Created by Fabrizio Duroni on 09.09.20.
//  Copyright © 2020 Fabrizio Duroni. All rights reserved.
//

#import "GameFactory.h"

@implementation GameFactory

+ (Game *)make {
    return [[Game alloc] initWithPlayer1:[[Player alloc] initWithScore:Love] player2:[[Player alloc] initWithScore:Love]];
}

+ (Game *)makeUsingPlayer1Score: (Score)player1Score player2Score: (Score)player2Score {
    return  [[Game alloc] initWithPlayer1:[[Player alloc] initWithScore:player1Score] player2:[[Player alloc] initWithScore:player2Score]];
}

@end

File: tennis-game/python-tennis-game/GameFactory.py
from Game import Game
from Player import Player
from Score import Score


class GameFactory:
    @staticmethod
    def make() -> Game:
        return Game(Player(Score.Love), Player(Score.Love))

    @staticmethod
    def make_using(score1: Score, score2: Score) -> Game:
        return Game(Player(score1), Player(score2))
